backend/src/main/kotlin/com/ci5644/trade/controllers/api/OfferController.kt

```diff
@@ -0,0 +1,171 @@
+
+import com.ci5644.trade.services.card.OfferService
+import com.ci5644.trade.dto.OfferDto
+import com.ci5644.trade.models.card.OfferStatus
+import com.ci5644.trade.config.SecurityConstants
+import com.ci5644.trade.config.JWT.JWTSecurityUtils
+import com.ci5644.trade.exceptions.runtime.OfferNotFoundException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import org.springframework.security.core.AuthenticationException
+import org.springframework.data.domain.Pageable
+import org.springframework.data.domain.Page
+
+
+
+
+/**
+ * Controller that contains all the endpoints related to offer management.
+ */
+
+@RestController
+@RequestMapping("/api/offer")
+class OfferController(private val offerService: OfferService) {
+
+    /**
+     * Creates a new offer.
+     *
+     * @param requestBody A map containing the request body with the keys 'cardOffer' for the card to offer and 'cardReceive' for the card to receive.
+     * @return A response entity containing a json with the created offer.
+     */
+    @PostMapping()
+    fun createOffer(
+        @CookieValue(name = SecurityConstants.AUTH_COOKIE_NAME) authCookie: String,
+        @RequestBody requestBody: Map<String, Int>
+    ): ResponseEntity<*> {
+        return try {
+            val username = JWTSecurityUtils.getAuthUserFromJWT(authCookie)
+            val cardOffer = requestBody["cardOffer"]
+            val cardReceive = requestBody["cardReceive"]
+    
+            if (cardOffer == null || cardReceive == null || cardOffer == cardReceive ) {
+                ResponseEntity.status(HttpStatus.BAD_REQUEST).body("La carta a ofrecer y la carta a recibir deben ser proporcionadas")
+            }
+    
+            offerService.createOffer(username, cardOffer!!, cardReceive!!)
+            ResponseEntity.ok("Offer created")
+        } catch (e: IllegalArgumentException) {
+            ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.message)
+        } catch (e: Exception) {
+            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.message)
+        }
+    }
+    
+
+    /* 
+    * Gets a list of offers with pending 
+    *
+    *@param authCookie The JWT cookie
+    * @return A list of offers with PENDING status
+    */
+    @GetMapping()
+    fun getOffers(
+        @CookieValue(name = SecurityConstants.AUTH_COOKIE_NAME) authCookie: String,
+        @RequestParam page: Int
+    ): ResponseEntity<*> {
+        return try {
+            val username = JWTSecurityUtils.getAuthUserFromJWT(authCookie) 
+
+            if (page < 0) {
+                return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("La página debe ser un número entero mayor o igual a 0")
+            } 
+
+            ResponseEntity.ok(offerService.getAvailableOffers(username, page))
+        } catch (e: NoSuchElementException) {
```

Comment:
if you are doing the same in both exceptions you can do 
} catch (e: NoSuchElementException | OfferNotFoundException) {